### ISSUE TITLE
[clang-format] BinPackParameters/BinPackArguments' data type is not align with document, BinPackParametersStyle <> boolean

### ISSUE THREAD

**Issue by radiocom** [llvm/llvm-project#117830]
I want my function defines tobe all formated to this:
```
returnType callFunction(
    typeA a,
    typeB b,
    typeC c,
    typeD d);

typedef returnType (*functionPointerName) (
    _In_ inType inVar);

void callFunction(
    typeA very_short_a);

returnType callFunction();
```

According to these documents, I use -style=file. and in my directory I create .clang-format file with all configurations.
https://github.com/llvm/llvm-project/blob/main/clang/docs/ClangFormatStyleOptions.rst
https://clang.llvm.org/docs/ClangFormatStyleOptions.html
```
AllowAllParametersOfDeclarationOnNextLine: true
BinPackParameters: AlwaysOnePerLine 
ColumnLimit: 130
```

This config file failed with error:
It seems BinPackParameters accepts a boolean value rather than the BinPackParametersStyle mentioned in our documentations.
```
.clang-format:25:20: error: invalid boolean
BinPackParameters: AlwaysOnePerLine
```

**Comment by owenca** [comment]
It was changed to an `enum` in 7c3237d778572931ff097e81df43d0bce9d1d4f8.

**Comment by radiocom** [comment]
> It was changed to an `enum` in [7c3237d](https://github.com/llvm/llvm-project/commit/7c3237d778572931ff097e81df43d0bce9d1d4f8).

Cool, thanks for your excellent work. but it seems not released even in latest 19.1.4. 
Could you please provide me a pipeline with new clang-format.exe that contains your change?
Thanks a lot.

**Comment by owenca** [comment]
> > It was changed to an `enum` in [7c3237d](https://github.com/llvm/llvm-project/commit/7c3237d778572931ff097e81df43d0bce9d1d4f8).
> 
> Cool, thanks for your excellent work. but it seems not released even in latest 19.1.4. Could you please provide me a pipeline with new clang-format.exe that contains your change? Thanks a lot.

I only reviewed and merged the patch, which was authored by another contributor. It's available in the main branch and will be released in LLVM 20.
